add Crossplane CLI convert ref docs (#694)
crossplane/docs: content/master/cli/command-reference.md

<!-- vale Crossplane.Spelling = YES -->
{{< /table >}}

### beta convert

As Crossplane evolves, its APIs and resources may change. To help with the 
migration to the new APIs and resources, the `crossplane beta convert` command

> converts a Crossplane resource to a new version or kind.

Reviewer: ```suggestion
converts an existing Crossplane resource to a new version or kind.
```

Author: This is the only suggestion I didnt use. My reasoning is that the resource does not have to exist somewhere, it can be just a yaml file laying around. But feel free to change if I am getting the meaning wrong

Reviewer: works for me, thanks for the clarification

> Supported conversions:
> * ControllerConfig to DeploymentRuntimeConfig
> * P&T Composition to Function Pipeline Composition

Reviewer: ```suggestion
The Crossplane CLI supported the following conversions:
* [ControllerConfig]({{<ref "../concepts/providers#controller-configuration">}}) to [DeploymentRuntimeConfig]({{<ref "../concepts/providers#runtime-configuration">}})
* [Composition patch and transforms]({{ ref "concepts/compositions/changing-resource-fields">}}) to [Function Pipeline Composition]({{< ref "../concepts/compositions#use-composition-functions">}})
```
add links to the features

> The command argument is a YAML file containing a single Crossplane resource. If
> not specified or `-`, it uses stdin. The command outputs the converted
> resource to stdout or a file.

Reviewer: ```suggestion
The command argument is a YAML file containing a single Crossplane resource.  
Don't provide a file argument or use `-` to use stdin.   

The command outputs the converted resource to stdout or a file.
```

Suggested tweak to make it more straight forward.

#### beta convert deployment-runtime

The `crossplane beta convert deployment-runtime` command converts a 
ControllerConfig to a DeploymentRuntimeConfig.

#### Flags
{{< table "table table-sm table-striped">}}
| Short flag   | Long flag       | Description                                                                                 |
| ------------ | --------------- | ------------------------------                                                              |

> | `-o`         | `--output-file` | The file to write the generated DeploymentRuntimeConfig to. If not set, outputs to stdout.  |

Reviewer: ```suggestion
| `-o`         | `--output-file` | The file to write the generated DeploymentRuntimeConfig to. Outputs to stdout by default.  |
```

<!-- vale Crossplane.Spelling = YES -->
{{< /table >}}

#### beta convert pipeline-composition

The `crossplane beta convert pipeline-composition` command converts a

> P&T Composition to a Function Pipeline Composition.

Reviewer: ```suggestion
Composition patch and transform to a Composition Pipeline Function.
```

#### Flags
{{< table "table table-sm table-striped">}}
| Short flag   | Long flag         | Description                                                                                 |
| ------------ | ----------------- | ------------------------------                                                              |

> | `-o`         | `--output-file`   | The file to write the generated DeploymentRuntimeConfig to. If not set, outputs to stdout.  |

Reviewer: ```suggestion
| `-o`         | `--output-file`   | The file to write the generated DeploymentRuntimeConfig to. Outputs to stdout by default.  |
```

> | `-f`         | `--function-name` | FunctionRefName to use. Defaults to function-patch-and-transform.                           |

Reviewer: ```suggestion
| `-f`         | `--function-name` | `functionRef.name` to use. Defaults to the name "function-patch-and-transform".                           |
```
It's `functionRef.name` right?

ControllerConfig to a DeploymentRuntimeConfig.

#### Flags
{{< table "table table-sm table-striped">}}
| Short flag   | Long flag       | Description                                                                                |
| ------------ | --------------- | ------------------------------                                                             |
| `-o`         | `--output-file` | The file to write the generated DeploymentRuntimeConfig to. Outputs to stdout by default.  |
<!-- vale Crossplane.Spelling = YES -->
{{< /table >}}

#### beta convert `pipeline-composition`

The `crossplane beta convert pipeline-composition` command converts a
Composition patch and transform to a Composition Pipeline Function.

#### Flags
{{< table "table table-sm table-striped">}}
| Short flag   | Long flag         | Description                                                                                |
| ------------ | ----------------- | ------------------------------                                                             |
| `-o`         | `--output-file`   | The file to write the generated DeploymentRuntimeConfig to. Outputs to stdout by default.  |

> | `-f`         | `--function-name` | `functionRef.name` to use. Defaults to name "function-patch-and-transform".                |

Reviewer: ```suggestion
| `-f`         | `--function-name` | `functionRef.name` to use. Defaults to name "function-patch-and-transform."                |
```

This should fix your weird Vale error with the wrong line being referenced.
